src/server/game/Entities/Player/Player.cpp

```diff
@@ -1710,12 +1710,9 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
             return true;
         }
 
-        if (!(options & TELE_TO_NOT_UNSUMMON_PET))
```

Comment:
removing this option is hack

Reply:
Thanks for pointing it out. The change looked a bit odd to me, but I didn't know why.

Reply:
I thought that might be too easy, but pets definitely don't teleport correctly.

The calling method is structure as such:
```
ToPlayer()->TeleportTo(target, TELE_TO_NOT_LEAVE_TRANSPORT | TELE_TO_NOT_LEAVE_COMBAT | TELE_TO_NOT_UNSUMMON_PET | (casting ? TELE_TO_SPELL : 0));
```
in Unit.cpp.  For hearthing, portals and item teleports it appears to evaluate to 30 consistently.  I could change that condition to check ```if (!(options == 30 ...```, which would correct this.

Although I don't quite follow the usage of the enum in ```if (!(options & TELE_TO_NOT_UNSUMMON_PET))```
Is that just to act as a label?

Reply:
This is meant to not unsummon any pets when doing a short distance teleport like a blink

Reply:
I see.  I did some reading on bit masks and the condition makes a lot more sense now, although I'm still not convinced it's necessary.  The line immediately following the condition is checking pet proximity and blocking the unsummon if it's within range of the new location.  When I hearth from outside the inn to inside the inn on my warlock, my imp just turns around and runs over to me.  No un/resummon.

It seems like the only case where this might be useful is when the pet is.
1. Out of range.
2. Not intended to be resummoned after a teleport.

I can't think of a single use case where that would apply.

Although if you're that intent on keeping the condition, the correct change would probably be to remove the enum from the calling method in Unit.cpp.  That should be removed either way.  I'll do some more digging to see if I can find any hidden side effects, but I've been driving this for a few of days now and it looks good.